ParkingZoneApp/Services/ParkingSlotService.cs

```diff
@@ -61,5 +60,22 @@ public IQueryable<ParkingSlot> FilterParkingSlot(IQueryable<ParkingSlot> query,
             }
             return query;
         }
+
+        public IEnumerable<Reservation> GetAllReservationsByParkingZoneId(int parkingZoneId)
```

Comment:
try to use LINQ here instead of long and inefficient lines of code. I tried gptand it is giving pretty nice code which looks more proffesional